### PR TITLE
Remove weights summary

### PR DESCRIPTION
**Pull request by shenoynikhil**
## What does this PR do?
Part of #12521 

This PR removes the `weights_summary` in the Trainer Class and the corresponding changes. Made the following edits,
- Removed Argument from Trainer Class [[Link](pytorch_lightning/trainer/trainer.py)]
- Removed functionality from Callback_Collector Class ([Link](pytorch_lightning/trainer/connectors/callback_connector.py))
- Removed test in `test_model_summary.py` ([Link](tests/utilities/test_model_summary.py))
- Removed Deprecation tests related to `weights_summary` ([Link](tests/deprecated_api/test_remove_1-7.py))

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 1]
```diff
@@ -162,21 +152,9 @@ def _configure_model_summary_callback(
...
-            )
-            if weights_summary not in ModelSummaryMode.supported_types():
-                raise MisconfigurationException(
-                    f"`weights_summary` can be None, {', '.join(ModelSummaryMode.supported_types())}",
-                    f" but got {weights_summary}",
-                )
-            max_depth = ModelSummaryMode.get_max_depth(weights_summary)
+        # If the user wants to configure a model summary callback, without explicitly passing it,
+        # to the callbacks list, we will create it with max depth of 1
+        max_depth = 1
```
Comment (shenoynikhil) [inline pytorch_lightning/trainer/connectors/callback_connector.py:157]: @rohitgr7 @akihironitta I have set the default value of max_depth as 1 if one does not pass a model summary callback but uses `enable_model_summary.` Let me know if this makes sense.

**Comment by shenoynikhil** [comment]
@akihironitta Can you also replace the PR mentioned against the issue in #12521 with this PR. The earlier version of PR was closed due to some issue.

File: pytorch_lightning/trainer/connectors/callback_connector.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/callback_connector.py:157]: Yes, that's correct because the previous default "top" `means max_depth=1`. But the model summary already has the `max_depth=1` as default, so this code can be removed (I would also remove the comment).

File: tests/utilities/test_model_summary.py [hunk 2]
```diff
@@ -139,19 +138,6 @@ def forward(self, inp):
...
-    """Test that invalid value for weights_summary raises an error."""
-    model = LightningModule()
-
-    with pytest.raises(
-        MisconfigurationException, match="`weights_summary` can be None, .* got temp"
-    ), pytest.deprecated_call(match="weights_summary=temp)` is deprecated"):
-        Trainer(weights_summary="temp")
-
-    with pytest.raises(ValueError, match="max_depth` can be .* got temp"):
-        ModelSummary(model, max_depth="temp")
```
Comment (awaelchli) [inline tests/utilities/test_model_summary.py:152]: This test part of the test should stay.

File: tests/utilities/test_model_summary.py [hunk 3]
```diff
@@ -139,19 +138,6 @@ def forward(self, inp):
...
 
 
-def test_invalid_weights_summary():
-    """Test that invalid value for weights_summary raises an error."""
-    model = LightningModule()
-
-    with pytest.raises(
-        MisconfigurationException, match="`weights_summary` can be None, .* got temp"
-    ), pytest.deprecated_call(match="weights_summary=temp)` is deprecated"):
-        Trainer(weights_summary="temp")
```
Comment (awaelchli) [inline tests/utilities/test_model_summary.py:149]: yes, this can be deleted, but not the part below

**Review by awaelchli — CHANGES_REQUESTED** [review]
Great work so far!

File: tests/utilities/test_model_summary.py (hunk 3 above)
Comment (shenoynikhil) [reply tests/utilities/test_model_summary.py:149]: Made the change, can you have a look

File: pytorch_lightning/trainer/connectors/callback_connector.py (hunk 1 above)
Comment (shenoynikhil) [reply pytorch_lightning/trainer/connectors/callback_connector.py]: @awaelchli Changed it, can you have a look?

File: tests/utilities/test_model_summary.py (hunk 2 above)
Comment (shenoynikhil) [reply tests/utilities/test_model_summary.py]: Changed it, please have a look.

**Review by awaelchli — APPROVED** [review]
Great!

**Review by rohitgr7 — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]
LGTM!

**Review by ananthsub — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by carmocca — COMMENTED** [review]
Gatekeeper
